plugins/repository-hdfs/build.gradle

```diff
@@ -154,29 +154,70 @@ if (fixtureSupported) {
 }
 
 boolean secureFixtureSupported = false;
+boolean vagrantSupported = false;
+boolean virtualBoxInstalled = false;
 if (fixtureSupported) {
   // Only do secure fixture support if the regular fixture is supported,
   // and if vagrant is installed. The ignoreExitValue on exec only matters
   // in cases where the command can be found and successfully started. In
   // situations where the vagrant command isn't able to be started at all
   // (it's not installed) then Gradle still throws ExecException.
-  ByteArrayOutputStream pipe = new ByteArrayOutputStream()
   try {
+    ByteArrayOutputStream pipe = new ByteArrayOutputStream()
     ExecResult runResult = exec {
       commandLine 'vagrant', '--version'
       standardOutput pipe
       ignoreExitValue true
     }
     String output = pipe.toString().trim()
     if (runResult.exitValue == 0) {
-      secureFixtureSupported = (output ==~ /Vagrant 1\.(8\.[6-9]|9\.[0-9])+/)
+      vagrantSupported = (output ==~ /Vagrant 1\.(8\.[6-9]|9\.[0-9])+/)
     } else {
       logger.warn("Could not read installed vagrant version:\n" + output)
     }
   } catch (org.gradle.process.internal.ExecException e) {
     logger.warn("Could not find vagrant: " + e.message)
     // Swallow error. Vagrant isn't installed. Leave secure fixture support off.
   }
+
+  // Finding that HOME needs to be set when performing vagrant updates
+  String homeLocation = System.getenv("HOME")
+  if (homeLocation == null) {
+    vagrantSupported = false
+  }
+
+  // Also check to see if virtualbox is installed. We need both vagrant
+  // and virtualbox to be installed for the secure environment to be
+  // enabled.
+  try {
+    ByteArrayOutputStream pipe = new ByteArrayOutputStream()
+    ExecResult runResult = exec {
+      commandLine 'vboxmanage', '--version'
```

Comment:
We really need to share this stuff...VagrantTestPlugin also does this.

Can we do this exactly once (for vagrant and vboxmanage each) and set a property on the project like `vagrantSupported`?  See for example in BuildPlugin how we check the java version etc in globalBuildInfo. This would also allow us to remove the tasks created by VagrantTestPlugin and just check the flag.